protocol/packages/currencies/build/main.rs

```diff
@@ -0,0 +1,148 @@
+use std::{
+    env,
+    fs::File,
+    io::{self, Write},
+    path::{Path, PathBuf},
+};
+
+use anyhow::{bail, Context as _, Result};
+
+use topology::{CurrencyDefinitions, Topology};
+
+use self::{currencies_tree::CurrenciesTree, either::Either, protocol::Protocol};
+
+mod convert_case;
+mod currencies_tree;
+mod either;
+mod protocol;
+mod sources;
+mod subtype_lifetime;
+
+const PROTOCOL_JSON: &str = "../../../build-configuration/protocol.json";
+
+const TOPOLOGY_JSON: &str = "../../../build-configuration/topology.json";
+
+const BUILD_REPORT: &str = "CURRENCIES_BUILD_REPORT";
+
+fn main() -> Result<()> {
+    for path in ["build/", PROTOCOL_JSON, TOPOLOGY_JSON] {
+        println!("cargo::rerun-if-changed={path}");
+    }
+
+    println!("cargo::rerun-if-env-changed={BUILD_REPORT}");
+
+    let output_directory: &Path = &PathBuf::from(
+        env::var_os("OUT_DIR").context("Cargo did not set `OUT_DIR` environment variable!")?,
+    );
+
+    let files_exist = IntoIterator::into_iter([PROTOCOL_JSON, TOPOLOGY_JSON])
+        .map(Path::new)
+        .map(Path::try_exists)
+        .try_fold(true, |all_exist, result| {
+            result
+                .map(|exists| all_exist && exists)
+                .context("Failed to check whether JSON descriptor file exists!")
+        })?;
+
+    let build_report = if let Some(report_file) = env::var_os(BUILD_REPORT) {
+        if files_exist {
```

Comment:
That is a really good point. Thank you for pointing it out!

Reply:
Looking a bit more into the context here, I actually do check that because my assumption is that if the environment variable is set, then the one setting it _expects_ that there should be a build step. That is why I error when that is not the case.